test/e2e/tests/resize_e2e_test.go

```diff
@@ -0,0 +1,241 @@
+		}()
+		validateDisk(diskInfo)
+
+		writeAndReadDisk(diskInfo)
+
+		offlineResizeDisk(diskInfo)
+
+		onlineResizeDisk(diskInfo)
+	})
+})
+
+func writeAndReadDisk(di *DiskInfo) {
+	instance := di.TestCtx.Instance
+	volName := di.Name
+
+	// Stage Disk
+	stageDir := filepath.Join("/tmp/", volName, "stage")
+	err := testutils.MkdirAll(instance, stageDir)
+	Expect(err).To(BeNil(), "Mkdir failed with error")
+	defer func() {
+		// delete remote directory
+		fp := filepath.Join("/tmp/", volName)
+		err = testutils.RmAll(instance, fp)
+		Expect(err).To(BeNil(), "Failed to delete remote directory")
+	}()
+
+	err = di.TestCtx.Client.NodeStageVolume(di.Volume.GetVolumeId(), stageDir, di.Volume.GetVolumeContext())
+	Expect(err).To(BeNil(), "NodeStageVolume failed with error")
+	defer func() {
+		// Unstage Disk
+		err = di.TestCtx.Client.NodeUnstageVolume(di.Volume.GetVolumeId(), stageDir)
+		Expect(err).To(BeNil(), "NodeUnstageVolume failed with error")
+	}()
+
+	// Mount Disk
+	publishDir := filepath.Join("/tmp/", volName, "mount")
+	err = testutils.MkdirAll(instance, publishDir)
+	Expect(err).To(BeNil(), "Mkdir failed with error")
+
+	err = di.TestCtx.Client.NodePublishVolume(di.Volume.GetVolumeId(), stageDir, publishDir, di.Volume.GetVolumeContext())
+	Expect(err).To(BeNil(), "NodePublishVolume failed with error")
+
+	// Write a file
+	err = testutils.ForceChmod(instance, publishDir, "777")
+	Expect(err).To(BeNil(), "Chmod failed with error")
+
+	testFileContents := "test"
+	testFile := filepath.Join(publishDir, "testfile")
+	err = testutils.WriteFile(instance, testFile, testFileContents)
+	Expect(err).To(BeNil(), "Failed to write file")
+
+	// Unmount Disk
+	err = di.TestCtx.Client.NodeUnpublishVolume(di.Volume.GetVolumeId(), publishDir)
+	Expect(err).To(BeNil(), "NodeUnpublishVolume failed with error")
+
+	// Mount disk somewhere else
+	secondPublishDir := filepath.Join("/tmp/", volName, "secondmount")
+	err = testutils.MkdirAll(instance, secondPublishDir)
+	Expect(err).To(BeNil(), "Mkdir failed with error")
+
+	err = di.TestCtx.Client.NodePublishVolume(di.Volume.GetVolumeId(), stageDir, secondPublishDir, di.Volume.GetVolumeContext())
+	Expect(err).To(BeNil(), "NodePublishVolume failed with error")
+
+	// Read File
+	err = testutils.ForceChmod(instance, secondPublishDir, "777")
+	Expect(err).To(BeNil(), "Chmod failed with error")
+
+	secondTestFile := filepath.Join(secondPublishDir, "testfile")
+	readContents, err := testutils.ReadFile(instance, secondTestFile)
+	Expect(err).To(BeNil(), "ReadFile failed with error")
+	Expect(strings.TrimSpace(string(readContents))).To(Equal(testFileContents))
+
+	// Unmount Disk
+	err = di.TestCtx.Client.NodeUnpublishVolume(di.Volume.GetVolumeId(), secondPublishDir)
+	Expect(err).To(BeNil(), "NodeUnpublishVolume failed with error")
+}
+
+func onlineResizeDisk(di *DiskInfo) {
+	instance := di.TestCtx.Instance
+	volName := di.Name
+
+	// Stage Disk
```

Comment:
The step of stage(defer unstage), publish(defer unpublish) is a common theme in tests.
We can perhaps make a function StageAndMount() which returns a cleanup function. Saves duplicate code and keeps the code logic in one place to make any changes. Thoughts?

```
StageAndPublish() {
    // stage
    // publish
    return cleanup_func() {
      // unpublish 
      //. unstage
    }
}
```

now we can write tests
```
onlineresizetest() {
    cleanup = StageAndPublish()
    defer cleanup()
    // resize test cases.
}
```
```
writeAndReadDisk() {
    cleanup = StageAndPublish()
    defer cleanup()
    // write tests
    // Publish to second dir
    // defer unpublish()
    // more write tests
}
```

Reply:
I like this idea, implemented! 2 things worth adding:
1. previously I was risk averse to helper methods b/c default gingko doesn't print out full stack trace, so it was difficult to debug when errors occurred. I just discovered this new flag option "-trace" that solves this issue, so now it's much easier :) 

2. doing it this way ~slightly~ changes the test where the filestore instance is published to 2 directories simultaneously. I don't think this affects the test in any meaningful way.